src/GitHub.Exports/Services/UsageTracker.cs

```diff
@@ -0,0 +1,251 @@
+        readonly string storePath;
+        UsageStore usage;
+
+        [ImportingConstructor]
+        public UsageTracker(
+            IProgram program,
+            Lazy<IConnectionManager> connectionManager,
+            IPackageSettings userSettings,
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        {
+            var componentModel = serviceProvider.GetService(typeof(SComponentModel)) as IComponentModel;
+
+            this.connectionManager = connectionManager;
+            this.userSettings = userSettings;
+            this.client = componentModel?.DefaultExportProvider.GetExportedValue<IMetricsService>();
+            this.timer = new DispatcherTimer(
+                TimeSpan.FromMinutes(1),
+                DispatcherPriority.Background,
+                TimerTick,
+                Dispatcher.CurrentDispatcher);
+            this.storePath = System.IO.Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+                program.ApplicationName,
+                StoreFileName);
+            this.usage = LoadUsage();
+
+            userSettings.PropertyChanged += (s, e) =>
+            {
+                if (e.PropertyName == nameof(userSettings.CollectMetrics))
+                {
+                    UpdateTimer(false);
+                }
+            };
+
+            UpdateTimer(true);
+        }
+
+        public void IncrementLaunchCount()
+        {
+            ++usage.Model.NumberOfStartups;
+            ++usage.Model.NumberOfStartupsWeek;
+            ++usage.Model.NumberOfStartupsMonth;
+            SaveUsage(usage);
+        }
+
+        public void IncrementCloneCount()
+        {
+            ++usage.Model.NumberOfClones;
+            SaveUsage(usage);
+        }
+
+        public void IncrementCreateCount()
+        {
+            ++usage.Model.NumberOfReposCreated;
+            SaveUsage(usage);
+        }
+
+        public void IncrementPublishCount()
+        {
+            ++usage.Model.NumberOfReposPublished;
+            SaveUsage(usage);
+        }
+
+        public void IncrementOpenInGitHubCount()
+        {
+            ++usage.Model.NumberOfOpenInGitHub;
+            SaveUsage(usage);
+        }
+
+        public void IncrementLinkToGitHubCount()
+        {
+            ++usage.Model.NumberOfLinkToGitHub;
+            SaveUsage(usage);
+        }
+
+        public void IncrementCreateGistCount()
+        {
+            ++usage.Model.NumberOfGists;
+            SaveUsage(usage);
+        }
+
+        public void IncrementUpstreamPullRequestCount()
+        {
+            ++usage.Model.NumberOfClones;
+            SaveUsage(usage);
+        }
+
+        UsageStore LoadUsage()
+        {
+            var result = File.Exists(storePath) ?
+                SimpleJson.DeserializeObject<UsageStore>(File.ReadAllText(storePath)) :
+                new UsageStore { Model = new UsageModel() };
```

Comment:
This whole thing is going to need exception handling, there's multiple things that can go wrong here. I'd like to follow the same pattern that https://github.com/github/VisualStudio/blob/master/src/GitHub.VisualStudio/Services/ConnectionManager.cs has - delegates for IO methods that can be mocked later on by our unit tests, a separate method that ensures the cache file exists, and splitting up reading/writing from serializing/deserializing, since one can fail because of the OS but the latter can fail because we break the format (and we can catch that in unit tests)

Note: `ConnectionManager` is not checking for IO exceptions at the moment. It's never failed because that would require two VS instances running at the same time and accessing `ConnectionManager` at exactly the same time, which is unlikely to happen (most cache accesses are user-initiated when logging in, so a user would have to login at the same time in two VS instances, unlikely). However, metrics are timer based which means the likelihood of two VS instances racing for the same file access could potentially be higher, so we need to watch out for this. Any crashes here will crash VS.